audiophiler/__init__.py

```diff
@@ -225,25 +263,45 @@ def get_harold(uid, auth_dict=None):
 
                 return get_file_s3(s3_bucket, harold_file_hash)
 
+    return "Permission denied", 403
+
+@app.route("/lock", method=["POST"])
+@auth.oidc_auth('default')
+@audiophiler_auth
+def toggle_tour_mode(auth_dict=None):
+    is_rtp = ldap_is_rtp(auth_dict["uid"])
+    is_eboard = ldap_is_eboard(auth_dict["uid"])
+    if not is_rtp or not is_eboard:
+        return "Permisssion Denied", 403
 
+    admin_query = Tour.query.first()
+    admin_query.tour_lock = not admin_query.tour_lock
```

Comment:
This shouldn't be a stateless toggle, since your button has state. This breaks down when in the case of multiple users.

Consider if A loads the page, and the button shows that it is locked.  THen B loads the page, and presses the button, unlocking it. Then A presses the button, which will now _relock_ it, rather than unlocking.

You've made a race condition.